osgp/protocol-adapter-iec60870/osgp-protocol-adapter-iec60870/src/main/java/org/opensmartgridplatform/adapter/protocol/iec60870/infra/messaging/OsgpRequestMessageSender.java

```diff
@@ -0,0 +1,54 @@
+/**
+ * Copyright 2020 Smart Society Services B.V.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ */
+package org.opensmartgridplatform.adapter.protocol.iec60870.infra.messaging;
+
+import javax.jms.JMSException;
+import javax.jms.Message;
+import javax.jms.ObjectMessage;
+import javax.jms.Session;
+
+import org.opensmartgridplatform.shared.infra.jms.Constants;
+import org.opensmartgridplatform.shared.infra.jms.RequestMessage;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Qualifier;
+import org.springframework.jms.core.JmsTemplate;
+import org.springframework.jms.core.MessageCreator;
+import org.springframework.stereotype.Component;
+
+@Component(value = "protocolIec60870OutboundOsgpCoreRequestsMessageSender")
+public class OsgpRequestMessageSender {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(OsgpRequestMessageSender.class);
+
+    @Autowired
+    @Qualifier("protocolIec60870OutboundOsgpCoreRequestsJmsTemplate")
+    private JmsTemplate jmsTemplate;
+
+    public void send(final RequestMessage requestMessage, final String messageType) {
+        LOGGER.info("Sending request message to OSGP.");
+
+        this.jmsTemplate.send(new MessageCreator() {
+
+            @Override
+            public Message createMessage(final Session session) throws JMSException {
+                final ObjectMessage objectMessage = session.createObjectMessage(requestMessage);
+                objectMessage.setJMSType(messageType);
+                objectMessage.setStringProperty(Constants.ORGANISATION_IDENTIFICATION,
+                        requestMessage.getOrganisationIdentification());
+                objectMessage.setStringProperty(Constants.DEVICE_IDENTIFICATION,
+                        requestMessage.getDeviceIdentification());
+
+                return objectMessage;
+            }
```

Comment:
we could do without the anonymous inner class:
this.jmsTemplate.send(session -> {
    final ObjectMessage objectMessage = session.createObjectMessage(requestMessage);
    objectMessage.setJMSType(messageType);
    objectMessage.setStringProperty(Constants.ORGANISATION_IDENTIFICATION,
            requestMessage.getOrganisationIdentification());
    objectMessage.setStringProperty(Constants.DEVICE_IDENTIFICATION, requestMessage.getDeviceIdentification());

    return objectMessage;
});

Reply:
I replaced the anonymous class by a lambda expression.
It may be better if we figure out if one form or the other is a preferred way of handling things.
Without a declared preference this might just lead to changing "in the other direction" in some other pull request.